cli/cmd/staking.go

```diff
@@ -81,85 +121,176 @@ func (c createValConfig) ConsensusPubKey() (*ecdsa.PublicKey, error) {
-	if !ok {
-		return errors.New("chain metadata not found")
+	eth, cprov, backend, err := setupClients(cfg.evmRPC, cfg.network, operatorPriv)
+	if err != nil {
+		return err
 	}
 
-	ethCl, err := ethclient.Dial(chainMeta.Name, cfg.Network.Static().ExecutionRPC())
-	if err != nil {
+	// check if we already have an existing validator
+	if _, ok, err := cprov.SDKValidator(ctx, opAddr); err != nil {
 		return err
+	} else if ok {
+		return &CliError{
+			Msg:     "Operator address already a validator: " + opAddr.Hex(),
+			Suggest: "Ensure correct operator address",
+		}
 	}
 
-	cprov, err := provider.Dial(cfg.Network)
+	contract, err := bindings.NewStaking(common.HexToAddress(predeploys.Staking), backend)
 	if err != nil {
 		return err
 	}
 
-	if _, ok, err = cprov.SDKValidator(ctx, opAddr); err != nil {
+	// only check if validator is on allow list if the allow list is enabled
+	bindOpts := &bind.CallOpts{Context: ctx}
+	if enabled, err := contract.IsAllowlistEnabled(bindOpts); err != nil {
 		return err
-	} else if ok {
+	} else if enabled {
+		if ok, err := contract.IsAllowedValidator(bindOpts, opAddr); err != nil {
+			return err
+		} else if !ok {
+			return &CliError{
+				Msg:     "Operator address not allowed to create validator: " + opAddr.Hex(),
+				Suggest: "Contact Omni team to be included in validator allow list",
+			}
+		}
+	}
+
+	bal, err := eth.EtherBalanceAt(ctx, opAddr)
+	if err != nil {
+		return err
+	} else if bal <= float64(cfg.selfDelegation) {
 		return &CliError{
-			Msg:     "Operator address already a validator: " + opAddr.Hex(),
-			Suggest: "Ensure correct operator address",
+			Msg:     fmt.Sprintf("Operator address has insufficient balance=%.2f OMNI, address=%s", bal, opAddr),
+			Suggest: "Fund the operator address with sufficient OMNI for self-delegation and gas",
 		}
 	}
 
-	backend, err := ethbackend.NewBackend(chainMeta.Name, chainID, chainMeta.BlockPeriod, ethCl, opPrivKey)
+	txOpts, err := backend.BindOpts(ctx, opAddr)
+	if err != nil {
+		return err
+	}
+	txOpts.Value = new(big.Int).Mul(umath.NewBigInt(cfg.selfDelegation), big.NewInt(params.Ether)) // Send self-delegation
+	consPubkey, err := cfg.consensusPublicKey()
 	if err != nil {
 		return err
 	}
 
-	contract, err := bindings.NewStaking(common.HexToAddress(predeploys.Staking), backend)
+	tx, err := contract.CreateValidator(txOpts, crypto.CompressPubkey(consPubkey))
+	if err != nil {
+		return errors.Wrap(err, "create validator")
+	}
+
+	rec, err := backend.WaitMined(ctx, tx)
+	if err != nil {
+		return errors.Wrap(err, "wait mined")
+	}
+
+	link := fmt.Sprintf("https://%s.omniscan.network/tx/%s", cfg.network, tx.Hash().Hex())
+	log.Info(ctx, "🎉 Create-validator transaction sent and included on-chain", "link", link, "block", rec.BlockNumber.Uint64())
+
+	return nil
+}
+
+type delegateConfig struct {
+	eoaConfig
+	amount uint64
+	self   bool
+}
+
+func (d delegateConfig) validate() error {
+	if d.amount < minDelegationIncrease {
```

Comment:
should probably add `if !d.self return "only self-delegation supported, please provide --self `